packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-ciemss-drilldown.vue

```diff
@@ -332,12 +332,11 @@
 								? getChartAnnotationsByChartId(activeChartSettings?.id ?? '')
 								: undefined
 						"
-						@update-settings-color="onColorChange"
 						:active-settings="activeChartSettings"
 						:generate-annotation="generateAnnotation"
-						@update-settings-scale="updateChartSettingsScale(activeChartSettings?.id as string, $event)"
+						@update-settings="findAndUpdateChartSettingsById(activeChartSettings?.id as string, $event)"
```

Comment:
I don't see the need of it since we can just simply use more generalized method `findAndUpdateChartSettingsById` with only id and the payload.

Reply:
Because you need to do a check to see if `activeChartSettings?.id` exist, and type it as a `string`.
When `activeChartSettings` is known and created by the composable.
The idea of reading a proprety from a composable to pass it back to a method... Feels a bit weird. Like what the point of the composable if it's not to have to handle those things and just change the state via method that can deal with a lack of id or not, etc.

Reply:
Ok I just added `updateActiveChartSettings` method to the composable for more convenience.